### ISSUE TITLE
Node cannot create more than one SparkContext per require

### ISSUE THREAD

**Issue by doronrosenberg** [EclairJS/eclairjs-node#109]
```
var eclairjs = require('eclairjs');

var sc1 = new eclairjs.SparkContext("local[*]", "Job 1");
var sc2 = new eclairjs.SparkContext("local[*]", "Job 2");

var p1 = sc1.appName();
var p2 = sc2.appName();

Promise.all([p1,p2]).then(function(results) {
  console.log("Success:", results);
})
```

Will output `Job 1` for both.

Working on this at https://github.com/EclairJS/eclairjs-node/tree/kernelscoping

**Comment by doronrosenberg** [comment]
@bpburns 

The current branch changes will require an API change and calls will look this:

```
var eclairjs = require('eclairjs');

var spark = new eclairjs();

var sc = new spark.SparkContext("local[*]", "foo");
```

**Comment by bpburns** [comment]
@doronrosenberg 

This doesn't look to bad.

**Comment by doronrosenberg** [comment]
Merged into master, will be part of 0.7
